### PR TITLE
Rules removed before Comment Rules begin

### PR DESCRIPTION
**Pull request by Dylan203**
exclude name="Drupal.Commenting.FunctionComment.InvalidNoReturn
exclude name="Drupal.Commenting.FunctionComment.MissingParamType"

### CODE REVIEW

File: src/GraphQL/Execution/Executor.php [hunk 1]
```diff
@@ -113,10 +113,10 @@ class Executor implements ExecutorImplementation {
    * @param \GraphQL\Type\Schema $schema
    * @param \GraphQL\Language\AST\DocumentNode $document
    * @param \Drupal\graphql\GraphQL\Execution\ResolveContext $context
-   * @param $root
-   * @param $variables
-   * @param $operation
-   * @param $resolver
+   * @param string $root
+   * @param string $variables
```
Comment (klausi) [inline src/GraphQL/Execution/Executor.php]: should be an array according to the doc comment on the class variable where this is assigned to. Please check those

File: src/GraphQL/Execution/Executor.php [hunk 2]
```diff
@@ -113,10 +113,10 @@ class Executor implements ExecutorImplementation {
    * @param \GraphQL\Type\Schema $schema
    * @param \GraphQL\Language\AST\DocumentNode $document
    * @param \Drupal\graphql\GraphQL\Execution\ResolveContext $context
-   * @param $root
-   * @param $variables
-   * @param $operation
-   * @param $resolver
+   * @param string $root
```
Comment (klausi) [inline src/GraphQL/Execution/Executor.php]: same here, root is mixed on the class

File: src/GraphQL/Execution/Executor.php [hunk 3]
```diff
@@ -153,10 +153,10 @@ public function __construct(
...
+   * @param string $variables
```
Comment (klausi) [inline src/GraphQL/Execution/Executor.php]: array

File: src/GraphQL/Execution/Executor.php [hunk 4]
```diff
@@ -153,10 +153,10 @@ public function __construct(
    * @param \GraphQL\Type\Schema $schema
    * @param \GraphQL\Language\AST\DocumentNode $document
    * @param \Drupal\graphql\GraphQL\Execution\ResolveContext $context
-   * @param $root
-   * @param $variables
-   * @param $operation
-   * @param $resolver
+   * @param string $root
```
Comment (klausi) [inline src/GraphQL/Execution/Executor.php]: mixed

File: src/GraphQL/Execution/Executor.php [hunk 5]
```diff
@@ -153,10 +153,10 @@ public function __construct(
...
+   * @param string $variables
+   * @param object $operation
```
Comment (klausi) [inline src/GraphQL/Execution/Executor.php]: string

File: src/GraphQL/Execution/Executor.php [hunk 6]
```diff
@@ -153,10 +153,10 @@ public function __construct(
...
+   * @param object $resolver
```
Comment (klausi) [inline src/GraphQL/Execution/Executor.php]: callable

File: src/GraphQL/Execution/ExecutorFactory.php [hunk 7]
```diff
@@ -27,10 +27,10 @@ public function __construct(ContainerInterface $container) {
    * @param \GraphQL\Executor\Promise\PromiseAdapter $adapter
    * @param \GraphQL\Type\Schema $schema
    * @param \GraphQL\Language\AST\DocumentNode $document
-   * @param $root
+   * @param string $root
    * @param \Drupal\graphql\GraphQL\Execution\ResolveContext $context
-   * @param $variables
-   * @param $operation
+   * @param string $variables
```
Comment (klausi) [inline src/GraphQL/Execution/ExecutorFactory.php]: array

File: src/GraphQL/Execution/ExecutorFactory.php [hunk 8]
```diff
@@ -27,10 +27,10 @@ public function __construct(ContainerInterface $container) {
    * @param \GraphQL\Executor\Promise\PromiseAdapter $adapter
    * @param \GraphQL\Type\Schema $schema
    * @param \GraphQL\Language\AST\DocumentNode $document
-   * @param $root
+   * @param string $root
```
Comment (klausi) [inline src/GraphQL/Execution/ExecutorFactory.php]: mixed

File: src/GraphQL/Execution/ExecutorFactory.php [hunk 9]
```diff
@@ -27,10 +27,10 @@ public function __construct(ContainerInterface $container) {
...
    * @param \Drupal\graphql\GraphQL\Execution\ResolveContext $context
-   * @param $variables
-   * @param $operation
+   * @param string $variables
+   * @param object $operation
```
Comment (klausi) [inline src/GraphQL/Execution/ExecutorFactory.php]: string

File: src/GraphQL/Execution/FieldContext.php [hunk 10]
```diff
@@ -63,7 +63,7 @@ public function setContextLanguage($language) {
    *
    * @param string $name
    *   The name of the context.
-   * @param $value
+   * @param int $value
```
Comment (klausi) [inline src/GraphQL/Execution/FieldContext.php]: don't think this is int only, let's use mixed

File: src/GraphQL/Execution/ResolveContext.php [hunk 11]
```diff
@@ -108,7 +108,7 @@ public function getContextLanguage() {
   }
 
   /**
-   * @param $language
+   * @param string $language
    *
    * @return string
```
Comment (klausi) [inline src/GraphQL/Execution/ResolveContext.php]: should be `@return $this`

File: src/GraphQL/Execution/ResolveContext.php [hunk 12]
```diff
@@ -127,7 +127,7 @@ public function setContextLanguage($language) {
    *   The resolve info object.
    * @param string $name
    *   The name of the context.
-   * @param $value
+   * @param int $value
```
Comment (klausi) [inline src/GraphQL/Execution/ResolveContext.php]: can be anything, mixed

File: src/GraphQL/Resolver/Context.php [hunk 13]
```diff
@@ -26,8 +26,8 @@ class Context implements ResolverInterface {
   /**
    * Context constructor.
    *
-   * @param $name
-   * @param $default
+   * @param string $name
+   * @param int $default
```
Comment (klausi) [inline src/GraphQL/Resolver/Context.php]: mixed

File: src/GraphQL/Resolver/Map.php [hunk 14]
```diff
@@ -18,7 +18,7 @@ class Map implements ResolverInterface {
   /**
    * Map constructor.
    *
-   * @param $resolver
+   * @param object $resolver
```
Comment (klausi) [inline src/GraphQL/Resolver/Map.php]: let's use mixed as on the class comment

File: src/GraphQL/Resolver/Path.php [hunk 15]
```diff
@@ -44,8 +44,8 @@ class Path implements ResolverInterface {
   /**
    * Path constructor.
    *
-   * @param $type
-   * @param $path
+   * @param string $type
+   * @param string $path
```
Comment (klausi) [inline src/GraphQL/Resolver/Path.php]: mixed like on the class comment.

File: src/GraphQL/Resolver/ResolverInterface.php [hunk 16]
```diff
@@ -11,8 +11,8 @@ interface ResolverInterface {
   /**
    * Resolve values for the fields.
    *
-   * @param $value
-   * @param $args
+   * @param int $value
```
Comment (klausi) [inline src/GraphQL/Resolver/ResolverInterface.php]: mixed

File: src/GraphQL/Resolver/ResolverInterface.php [hunk 17]
```diff
@@ -11,8 +11,8 @@ interface ResolverInterface {
...
+   * @param string $args
```
Comment (klausi) [inline src/GraphQL/Resolver/ResolverInterface.php]: mixed

File: src/GraphQL/Resolver/Value.php [hunk 18]
```diff
@@ -19,7 +19,7 @@ class Value implements ResolverInterface {
   /**
    * Value constructor.
    *
-   * @param $value
+   * @param int $value
```
Comment (klausi) [inline src/GraphQL/Resolver/Value.php]: mixed

File: src/GraphQL/ResolverBuilder.php [hunk 19]
```diff
@@ -24,12 +24,12 @@ class ResolverBuilder {
   use DataFetcherTrait;
 
   /**
-   * @param $id
-   * @param $config
+   * @param int $id
```
Comment (klausi) [inline src/GraphQL/ResolverBuilder.php]: string

File: src/GraphQL/ResolverBuilder.php [hunk 20]
```diff
@@ -136,8 +136,8 @@ public function fromContext($name, $default = NULL) {
   }
 
   /**
-   * @param $value
-   * @param $default
+   * @param int $value
+   * @param object $default
```
Comment (klausi) [inline src/GraphQL/ResolverBuilder.php]: mixed

File: src/GraphQL/ResolverRegistry.php [hunk 21]
```diff
@@ -123,8 +123,8 @@ public function getTypeResolver($type) {
   }
 
   /**
-   * @param $value
-   * @param $args
+   * @param int $value
```
Comment (klausi) [inline src/GraphQL/ResolverRegistry.php]: mixed

File: src/GraphQL/ResolverRegistry.php [hunk 22]
```diff
@@ -135,8 +135,8 @@ protected function getRuntimeFieldResolver($value, $args, ResolveContext $contex
...
```
Comment (klausi) [inline src/GraphQL/ResolverRegistry.php]: mixed

File: src/GraphQL/ResolverRegistry.php [hunk 23]
```diff
@@ -135,8 +135,8 @@ protected function getRuntimeFieldResolver($value, $args, ResolveContext $contex
...
+   * @param string $args
```
Comment (klausi) [inline src/GraphQL/ResolverRegistry.php]: mixed

File: src/GraphQL/ResolverRegistry.php [hunk 24]
```diff
@@ -149,7 +149,7 @@ protected function resolveFieldDefault($value, $args, ResolveContext $context, R
   }
 
   /**
-   * @param $value
+   * @param int $value
```
Comment (klausi) [inline src/GraphQL/ResolverRegistry.php]: mixed

File: src/Plugin/DataProducerPluginManager.php [hunk 25]
```diff
@@ -77,7 +77,7 @@ public function __construct(
   }
 
   /**
-   * @param $id
+   * @param int $id
```
Comment (klausi) [inline src/Plugin/DataProducerPluginManager.php]: string

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php [hunk 26]
```diff
@@ -100,7 +100,7 @@ public function __construct(
   }
 
   /**
-   * @param $id
+   * @param int $id
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: string

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php [hunk 27]
```diff
@@ -135,8 +135,8 @@ public function cached($cached = TRUE) {
   /**
    * Resolve field value.
    *
-   * @param $value
-   * @param $args
+   * @param int $value
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: mixed

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php [hunk 28]
```diff
@@ -135,8 +135,8 @@ public function cached($cached = TRUE) {
...
+   * @param string $args
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: mixed

File: src/Plugin/GraphQL/DataProducer/Entity/EntityAccess.php [hunk 29]
```diff
@@ -32,8 +32,8 @@ class EntityAccess extends DataProducerPluginBase {
 
   /**
    * @param \Drupal\Core\Entity\EntityInterface $entity
-   * @param $operation
-   * @param $user
+   * @param string $operation
+   * @param string $user
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/Entity/EntityAccess.php]: \Drupal\Core\Session\AccountInterface

File: src/Plugin/GraphQL/DataProducer/Entity/EntityLoad.php [hunk 30]
```diff
@@ -128,9 +128,9 @@ public function __construct(
   }
 
   /**
-   * @param $type
-   * @param $id
-   * @param $language
+   * @param string $type
+   * @param int $id
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/Entity/EntityLoad.php]: string

File: src/Plugin/GraphQL/DataProducer/Images/ImageResourceUrl.php [hunk 31]
```diff
@@ -22,7 +22,7 @@
 class ImageResourceUrl extends DataProducerPluginBase {
 
   /**
-   * @param $derivative
+   * @param object $derivative
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/Images/ImageResourceUrl.php]: array

File: tests/src/Traits/DataProducerExecutionTrait.php [hunk 32]
```diff
@@ -10,7 +10,7 @@
 trait DataProducerExecutionTrait {
 
   /**
-   * @param $id
+   * @param int $id
```
Comment (klausi) [inline tests/src/Traits/DataProducerExecutionTrait.php]: string

File: tests/src/Traits/MockingTrait.php [hunk 33]
```diff
@@ -157,7 +157,7 @@ protected function mockSchema($id, $schema, array $extensions = []) {
   /**
    * Mock schema plugin manager.
    *
-   * @param $id
+   * @param int $id
```
Comment (klausi) [inline tests/src/Traits/MockingTrait.php]: string

**Review by klausi — CHANGES_REQUESTED** [review]
Thanks, please check all instances variables and where we should use mixed.

File: src/GraphQL/ResolverBuilder.php [hunk 34]
```diff
@@ -101,7 +101,7 @@ public function fromPath($type, $path, ResolverInterface $value = NULL) {
   }
 
   /**
-   * @param $value
+   * @param int $value
```
Comment (rthideaway) [inline src/GraphQL/ResolverBuilder.php]: `$value` is `mixed` type here

File: src/GraphQL/ResolverRegistry.php [hunk 35]
```diff
@@ -160,7 +160,7 @@ protected function getRuntimeTypeResolver($value, ResolveContext $context, Resol
...
```
Comment (rthideaway) [inline src/GraphQL/ResolverRegistry.php]: `$value` is `mixed` type here

File: src/GraphQL/ResolverRegistryInterface.php [hunk 36]
```diff
@@ -10,8 +10,8 @@
 interface ResolverRegistryInterface {
 
   /**
-   * @param $value
-   * @param $args
+   * @param int $value
```
Comment (rthideaway) [inline src/GraphQL/ResolverRegistryInterface.php]: `$value` is `mixed` type here

File: src/GraphQL/ResolverRegistryInterface.php [hunk 37]
```diff
@@ -10,8 +10,8 @@
...
+   * @param string $args
```
Comment (rthideaway) [inline src/GraphQL/ResolverRegistryInterface.php]: `$args` is `mixed` type here

File: src/GraphQL/ResolverRegistryInterface.php [hunk 38]
```diff
@@ -21,7 +21,7 @@ interface ResolverRegistryInterface {
   public function resolveField($value, $args, ResolveContext $context, ResolveInfo $info, FieldContext $field);
 
   /**
-   * @param $value
+   * @param int $value
```
Comment (rthideaway) [inline src/GraphQL/ResolverRegistryInterface.php]: `$value` is `mixed` type here

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php (hunk 28 above)
Comment (rthideaway) [reply src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: reminding that `$args` should be `mixed` type

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php [hunk 39]
```diff
@@ -167,8 +167,8 @@ public function resolve($value, $args, ResolveContext $context, ResolveInfo $inf
   }
 
   /**
-   * @param $value
-   * @param $args
+   * @param int $value
```
Comment (rthideaway) [inline src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: `mixed`

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php [hunk 40]
```diff
@@ -167,8 +167,8 @@ public function resolve($value, $args, ResolveContext $context, ResolveInfo $inf
...
+   * @param string $args
```
Comment (rthideaway) [inline src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: `mixed`

File: src/Plugin/GraphQL/DataProducer/DataProducerProxy.php [hunk 41]
```diff
@@ -279,8 +279,8 @@ protected function cacheRead($prefix) {
   }
 
   /**
-   * @param $prefix
-   * @param $value
+   * @param string $prefix
+   * @param int $value
```
Comment (rthideaway) [inline src/Plugin/GraphQL/DataProducer/DataProducerProxy.php]: `mixed`

**Review by rthideaway — CHANGES_REQUESTED** [review]
found couple of mistypes which need to be corrected

File: src/GraphQL/Resolver/Map.php [hunk 42]
```diff
@@ -18,7 +18,7 @@ class Map implements ResolverInterface {
   /**
    * Map constructor.
    *
-   * @param $resolver
+   * @param mixed $resolver
```
Comment (rthideaway) [inline src/GraphQL/Resolver/Map.php]: actully this should be `\Drupal\graphql\GraphQL\Resolver\ResolverInterface`

Comment (rthideaway) [reply src/GraphQL/Resolver/Map.php]: please also fix `$resolver` variable above, as it should be also the same type, and not `mixed`

**Review by rthideaway — CHANGES_REQUESTED** [review]
found two more issues, let's fix those and we're fine here

**Review by rthideaway — APPROVED** [review]
looks good to me, thanks!

**Pull request merged**